packages/camera/camera_web/lib/src/camera_service.dart

```diff
@@ -393,4 +394,68 @@ class CameraService {
         return DeviceOrientation.portraitUp;
     }
   }
+
+  /// Maps a [Uint] to a [CameraImageData].
+  CameraImageData getCameraImageDataFromBytes(
+    Uint8List bytes, {
+    required int height,
+    required int width,
+  }) {
+    return CameraImageData(
+      format: const CameraImageFormat(
+        ImageFormatGroup.jpeg,
+        raw: '',
+      ),
+      planes: <CameraImagePlane>[
+        CameraImagePlane(
+          bytes: bytes,
+          bytesPerRow: 0,
+        ),
+      ],
+      height: height,
+      width: width,
+    );
+  }
+
+  ///Returns frame at a specific time using video element
+  CameraImageData takeFrame(html.VideoElement videoElement) {
+    final List<String> widthPx = videoElement.style.width.split('px');
+    final List<String> heightPx = videoElement.style.height.split('px');
+    final String widthString =
+        widthPx.isNotEmpty ? widthPx.first : '${videoElement.videoWidth}';
+    final String heightString =
+        heightPx.isNotEmpty ? heightPx.first : '${videoElement.videoHeight}';
+    final int width = int.tryParse(widthString) ?? videoElement.videoWidth;
+    final int height = int.tryParse(heightString) ?? videoElement.videoHeight;
+
+    if (width == 0 || height == 0) {
+      throw Exception(
+        'Computed dimensions are zero: width=$width, height=$height',
+      );
+    }
+    final bool canUseOffscreenCanvas =
+        jsUtil.hasProperty(window!, 'OffscreenCanvas');
+    late html.ImageData imageData;
+    if (canUseOffscreenCanvas) {
+      final html.OffscreenCanvas canvas = html.OffscreenCanvas(width, height);
+      final html.OffscreenCanvasRenderingContext2D context =
+          canvas.getContext('2d')! as html.OffscreenCanvasRenderingContext2D;
+      context.drawImage(videoElement, 0, 0, width, height);
+      imageData = context.getImageData(0, 0, width, height);
+    } else {
+      final html.CanvasElement canvas =
+          html.CanvasElement(width: width, height: height);
+      final html.CanvasRenderingContext2D context = canvas.context2D;
+      context.drawImageScaled(videoElement, 0, 0, width, height);
+      imageData = context.getImageData(0, 0, width, height);
+      canvas.remove();
```

Comment:
It seems `canvas` was never added to the DOM, so `canvas.remove()` is probably unnecessary?